### DISCUSSION TITLE
Workspace package is not linked when depended upon from another package

### DISCUSSION THREAD

**Discussion by ackvf** [pnpm/pnpm#8182] [Q&A]
### tl;dr

**What is the correct way to depend on packages from within the monorepo?**

---

With this folder layout, where holograph-app and holograph-operator are our two apps, that both use single all encompassing UI/utils library `packages/holograph/`,
```
pnpm-workspace.yaml
.npmrc
package.json
apps/
	├─ holograph-app/
		├─ package.json
	├─ holograph-operator/
packages/
	├─ holograph/
	├─ eslint-config-holograph/
	├─ next-config/
```

and these files
`pnpm-workspace.yaml`
```
packages:
  - apps/*
  - packages/*
```

`.npmrc` _(tried with and without this file)_
```
link-workspace-packages=true    # also tried deep
prefer-workspace-packages=true
recursive-install=true
```

`package.json`
```
  "devDependencies": {
    "eslint-config-holograph": "*"
  }
```

both `apps/holograph-app/package.json` and `apps/holograph-operator/package.json`
```
  "dependencies": {
    "holograph": "*"
  },
  "devDependencies": {
    "eslint-config-holograph": "*",
    "next-config": "*",
  }
```

After installing from the root with `pnpm install`, I see this output
```
>pnpm install
Scope: all 11 workspace projects
Lockfile is up to date, resolution step is skipped
Packages: +1979
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
Progress: resolved 1979, reused 1974, downloaded 0, added 1979, done

devDependencies:
+ eslint 7.32.0
+ eslint-config-holograph 0.0.0 <- packages\eslint-config-holograph
+ turbo 1.13.4
+ typescript 5.4.5
...
```

**Which seems to understand the link for `eslint-config-holograph` correctly,**

HOWEVER,

when I browse the `holograph-app` or operator projects' files, I get unresolved dependencies in VSCode, and a failing `pnpm build` command too - which can't resolve dependencies.

_(ignore the first two unrelated errors)_
![image](https://github.com/pnpm/pnpm/assets/2313018/e45d54fa-2d63-4443-8398-932cddf07632)
![image](https://github.com/pnpm/pnpm/assets/2313018/dc80b45e-a231-4479-8f0e-d9f37556a934)

At some point I was also receiveing errors like these, but after several tries and configuration changes, I am no longer getting these.
```
eslint-config-holograph is not in the npm registry, or you have no permission to fetch it.
```

Adding this to the root `package.json` seems to have fixed it, but it doesn't feel right, given the `pnpm-workspace.yaml` and `.npmrc` configurations.

`package.json`
```
  "pnpm": {
    "overrides": {
      "eslint-config-holograph": "link:packages/eslint-config-holograph",
      "holograph": "link:packages/holograph",
      "next-config": "link:packages/next-config",
      "tailwind-config": "link:packages/tailwind-config",
      "tsconfig": "link:packages/tsconfig"
    }
  },
```

What is the correct way to solve this?
Meaning: Have a monorepo with several apps, all of them depending on other monorepo's packages?
